documentation/guides/migrating-from-v3-to-v4.md

```diff
@@ -0,0 +1,390 @@
+# Migrating from v3 to v4
+
+Polaris v4 contains mostly internal changes and has simple migration paths. This guide contains multiple steps on how to deal with our breaking changes in your application.
+
+**Note:** This migration guide is for v3 but may apply for v2 as well.
+
+- [Testing](#testing)
+- [API Changes](#api-changes)
+  - [AppProvider](#appprovider)
+    - [i18n](#i18n)
+  - [Navigation](#navigation)
+    - [iconBody](#icon-body)
+  - [ChoiceList](#choicelist)
+    - [title](#title)
+  - [Card](#card)
+    - [secondaryFooterAction](#secondaryfooteraction)
+  - [Select](#select)
+    - [groups](#groups)
+  - [Icon](#icon)
+    - [source](source)
+    - [untrusted](#untrusted)
+- [Removed Exports/Components](#removed-exportscomponents)
+  - [WithContext](#withcontext)
+  - [WithRef](#withref)
+  - [Autocomplete.ComboBox.TextField](#autocompletecomboboxtextfield)
+  - [Autocomplete.ComboBox.OptionList](#autocompletecomboboxoptionlist)
+  - [Navigation.UserMenu](#navigationusermenu)
+  - [Modal.Dialog](#modaldialog)
+  - [Tabs.Panel](#tabspanel)
+- [Dependencies](#dependencies)
+- [Typescript](#typescript)
+  - [Config](#config)
+  - [Types](#types)
+
+## Testing
+
+We've migrated to react new context API as well as restructured our entire context structure. To keep up to date using our test provider.
```

Comment:
```suggestion
We’ve migrated to [React’s new context API](https://reactjs.org/docs/context.html) as well as restructured Polaris’ entire context structure to keep up to date using the Polaris test provider.
```

This sentence isn't too easy to understand. I think there's a better way to make this clear but I don't have enough context.